core/src/processing/opengl/PSurfaceJOGL.java

```diff
@@ -273,6 +273,15 @@ protected void initGL() {
       } else if (PJOGL.profile == 2) {
         try {
           profile = GLProfile.getGL2ES2();
+
+          // workaround for https://jogamp.org/bugzilla/show_bug.cgi?id=1347
+          if (!profile.isHardwareRasterizer()) {
+            GLProfile hardware = GLProfile.getMaxProgrammable(true);
```

Comment:
Seems to give me the hardware profile - here are my debug printlns while I was testing the code above:

    Trying GL2ES2
    libGL error: MESA-LOADER: malformed or no PCI ID
    libGL error: unable to load driver: sun4i-drm_dri.so
    libGL error: driver pointer missing
    libGL error: failed to load driver: sun4i-drm
    Got GLProfile[GL2/GL2.sw]
    Not hardware
    Max programmable is GLProfile[GLES2/GLES2.hw]
    Picking max programmable over software